force-app/main/default/classes/LeadTriggerTest.cls

```diff
@@ -0,0 +1,312 @@
+
+        ldList.add(leadVP);
+        ldList.add(leadMgr);
+        ldList.add(leadExec);
+        ldList.add(leadAssist);
+
+        // perform test
+        Test.startTest();
+        insert ldList;
+        Test.stopTest();
+
+        // get values to check things are working as they should
+        List<Lead> ldListCheckVP = [select Id, Title from Lead where Id = :leadVP.Id];
+        List<Lead> ldListCheckMgr = [select Id, Title from Lead where Id = :leadMgr.Id];
+        List<Lead> ldListCheckExec = [select Id, Title from Lead where Id = :leadExec.Id];
+        List<Lead> ldListCheckAssist = [select Id, Title from Lead where Id = :leadAssist.Id];
+
+        // check results
+        System.assertEquals('Vice President', ldListCheckVP[0].Title, 'VP Title is not Vice President');
+        System.assertEquals('Manager', ldListCheckMgr[0].Title, 'Mgr Title is not Manager');
+        System.assertEquals('Executive', ldListCheckExec[0].Title, 'Exec Title is not Executive');
+        System.assertEquals('Assistant', ldListCheckAssist[0].Title, 'Assistant Title is not Assistant');
+    }
+
+    /*
+    * Test updating the titles to make sure the before update trigger fires and test pass.
+    */
+    @IsTest
+    static void testHandleTitleNormalization_Update() {
+
+        // creates leads        
+        List<Lead> ldList = new List<Lead>();
+
+        Lead leadVP = new Lead(FirstName='Test', LastName='Lead', Company='Test Company', Title='v.p.');
+        Lead leadMgr = new Lead(FirstName='Test', LastName='Lead', Company='Test Company', Title='manage');
+        Lead leadExec = new Lead(FirstName='Test', LastName='Lead', Company='Test Company', Title='chief');
+        Lead leadAssist = new Lead(FirstName='Test', LastName='Lead', Company='Test Company', Title='deputy');
+
+        ldList.add(leadVP);
+        ldList.add(leadMgr);
+        ldList.add(leadExec);
+        ldList.add(leadAssist);
+
+        insert ldList;
+
+        // get values to check things are working as they should
+        List<Lead> updatedLeads = new List<Lead>();
+
+        List<Lead> ldListCheckVP = [select Id, Title from Lead where Id = :leadVP.Id];
+        List<Lead> ldListCheckMgr = [select Id, Title from Lead where Id = :leadMgr.Id];
+        List<Lead> ldListCheckExec = [select Id, Title from Lead where Id = :leadExec.Id];
+        List<Lead> ldListCheckAssist = [select Id, Title from Lead where Id = :leadAssist.Id];
+        
+        ldListCheckVP[0].Title = 'vp';
+        ldListCheckMgr[0].Title = 'mgr';
+        ldListCheckExec[0].Title = 'exec';
+        ldListCheckAssist[0].Title = 'assist';
+       
+        updatedLeads.add(ldListCheckVP[0]);
+        updatedLeads.add(ldListCheckMgr[0]);
+        updatedLeads.add(ldListCheckExec[0]);
+        updatedLeads.add(ldListCheckAssist[0]);
+
+        // perform test
+        Test.startTest();
+        update updatedLeads;
```

Comment:
Try implementing the same suggestion I gave on the prior method regarding calling the method directly in the test and only doing a single query to assert the results